app/src/test/java/com/codesoom/assignment/controllers/TaskControllerTest.java

```diff
@@ -0,0 +1,130 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.application.TaskService;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class TaskControllerTest {
+    private static final String TASK_TITLE = "Test";
+    private static final String ANOTHER_TASK_TITLE = "Test2";
+    private static final String MODIFY_TASK_TITLE = "Modified";
+
+    private TaskController taskController;
+    private TaskService taskService;
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+        taskController = new TaskController(taskService);
+    }
+
+    /**
+     * Test about "GET /tasks" request.
+     * Check if task list is empty before add new task
+     */
+    @Test
+    void listBeforeAdding() {
+        assertThat(taskController.list()).isEmpty();
+    }
+
+    /**
+     * Test about "GET /tasks/:id" request with valid id.
+     * Check if the selected task has correct id and title.
+     */
+    @Test
+    void detailWithValidId() {
+        addTask(TASK_TITLE);
+
+        assertThat(taskController.detail(1L).getId()).isEqualTo(1L);
+        assertThat(taskController.detail(1L).getTitle()).isEqualTo(TASK_TITLE);
+    }
+
+    /**
+     * Test about "GET /tasks/:id" request with invalid id.
+     * Check retuning expected exception.
+     */
+    @Test
+    void detailWithInvalidId() {
+        assertThatThrownBy(() -> taskController.detail(100L))
+                .isInstanceOf(TaskNotFoundException.class);
+    }
```

Comment:
junit5 의 `assertThrows`를 사용해보면 어떨까요? 다음은 제가 작성한 테스트입니다. 이렇게 하면 어떤 점이 좋은가요?
```suggestion
    @Test
    void detailWithInvalidId() {
        Assertions.assertThrows(
            TaskNotFoundException.class,
            () -> taskController.detail(100L),
            "할 일을 찾을 수 없을 경우를 표현하는 예외가 던져져야 합니다"
        );
    }
```

Reply:
해당 테스트 코드의 목적이 잘 드러납니다. 피드백 감사합니다. 👍

Reply:
기존의 코드를 최대한 살리는(?) 방향으로 아래와 같이 구현해볼려고 하는데, @johngrib 님 의견은 어떠신가요? 
```
assertThatThrownBy(() -> taskController. detail(100L), "task를 찾을 수 없습니다.")
                        .isInstanceOf(TaskNotFoundException.class);
```

Reply:
저 메시지는 테스트가 실패했을 때 출력되는 것이 목표인가요? `assertThatThrownBy` 메소드의 주석을 읽고 저에게 설명해주세요.

Reply:
주석을 읽어보니 잘못된 목적으로 메시지가 설정되었습니다.
exception이 발생하지 않았을 때, 출력되게끔 설정되었습니다.
하지만 제 의도는 exception이 발생할 때, 출력되는 것이었습니다.

Reply:
메시지의 용도를 제가 잘못 이해한 것 같습니다.
해당 테스트가 실패할 때 출력되는 메시지인데, 테스트 자체를 설명하기 위해 메시지를 넣은 것 같습니다.
메시지가 위의 제안처럼 같이 변경되면 좋을 것 같습니다.

Reply:
직접 메소드 주석을 읽어본 일이 좋은 경험이 되었을 것이라 생각합니다. 축하드려요!